internal/client/kubernetessecret/secretexport/secret_export_resource.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright © 2021 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package secretexportclient
+
+import (
+	"net/url"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/transport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	secret "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster/secretexport"
+)
+
+const (
+	apiVersionAndGroup                 = "v1alpha1/clusters"
+	apiKind                            = "secretexports"
+	nameSpaces                         = "namespaces"
+	queryParamKeyManagementClusterName = "fullName.managementClusterName"
+	queryParamKeyProvisionerName       = "fullName.provisionerName"
+)
+
+// New creates a new namespace resource service API client.
+func New(transport *transport.Client) ClientService {
+	return &Client{Client: transport}
+}
+
+/*
+Client for namespace resource service API.
+*/
+type Client struct {
+	*transport.Client
+}
+
+// ClientService is the interface for Client methods
+type ClientService interface {
+	SecretExportResourceServiceCreate(request *secret.VmwareTanzuManageV1alpha1ClusterNamespaceSecretexportSecretExportRequest) (*secret.VmwareTanzuManageV1alpha1ClusterNamespaceSecretexportSecretExportResponse, error)
+
+	SecretExportResourceServiceDelete(fn *secret.VmwareTanzuManageV1alpha1ClusterNamespaceSecretexportFullName) error
+
+	SecretExportResourceServiceGet(fn *secret.VmwareTanzuManageV1alpha1ClusterNamespaceSecretexportFullName) (*secret.VmwareTanzuManageV1alpha1ClusterNamespaceSecretexportSecretExportResponse, error)
+}
+
+/*
+SecretExportResourceServiceCreate creates a secret export
+*/
+func (c *Client) SecretExportResourceServiceCreate(request *secret.VmwareTanzuManageV1alpha1ClusterNamespaceSecretexportSecretExportRequest) (*secret.VmwareTanzuManageV1alpha1ClusterNamespaceSecretexportSecretExportResponse, error) {
+	requestURL := helper.ConstructRequestURL(apiVersionAndGroup, request.SecretExport.FullName.ClusterName, nameSpaces, request.SecretExport.FullName.NamespaceName, apiKind).String()
+	secretResponse := &secret.VmwareTanzuManageV1alpha1ClusterNamespaceSecretexportSecretExportResponse{}
```

Comment:
can you check the variables naming and have it changed according to the context

Reply:
Updated PTAL